src/main/java/clzzz/helper/logic/parser/ParserUtil.java

```diff
@@ -94,6 +96,9 @@ public static DateOfBirth parseDateOfBirth(String dateOfBirth) throws ParseExcep
         if (!DateOfBirth.isValidDateOfBirth(trimmedDateOfBirth)) {
             throw new ParseException(DateOfBirth.MESSAGE_CONSTRAINTS);
         }
+        if (!DateTimeUtil.isValidDateString(trimmedDateOfBirth)) {
+            throw new ParseException(Messages.MESSAGE_INVALID_DATE);
+        }
```

Comment:
The naming doesn't make any sense, a valid date of birth is of course of a valid date. The sole purpose of having two checks is to have separate error messages.
Consider AB3's [Email](https://github.com/nus-cs2103-AY1920S2/addressbook-level3/blob/master/src/main/java/seedu/address/model/person/Email.java), which lists out possible errors without identifying them. 
Consider also our [Food](https://github.com/AY1920S2-CS2103-W15-4/main/blob/master/src/main/java/clzzz/helper/model/pet/Food.java), which checks both food name and amount but does not misleadingly call either check isValidFood().

Reply:
isValidDateOfBirth already uses LocalDate.parse with LENIENT to check the format, and then the date ranges are checked manually.
This is our own wrapper class, calling library class functions is inherently wasteful, the result of calling parse is ignored, so why not just call parse again with STRICT?

Reply:
ok then I will try that

Reply:
emm actually it doesn't work. cuz even if the date is in correct format and is valid, like 1/1/2020, strict will always throw an exception. if u really want to do that, we have to make user input follow dd/mm/yyyy strictly.

Reply:
uuuu
😭😭